### ISSUE TITLE
potential bugs

### ISSUE THREAD

**Issue by gd-zhang** [ikostrikov/pytorch-a2c-ppo-acktr#54]

**Comment by ikostrikov** [comment]
Hi,

cov_a and cov_g are implemented this way in order to be consistent the original KFAC code from OpenAI.

It trues that it can be implemented in a different way.

I multiply by batch_size because these gradients were averaged in the loss function. See:
https://github.com/openai/baselines/blob/master/baselines/acktr/kfac.py#L417

**Comment by gd-zhang** [comment]
Hmm ... I can understand why you multiply by batch.

But for compute_cov_a & compute_cov_g, I don't think you and original OpenAI code implement right. [https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/kfac](tf.contrib.kfac) is the official code for KFAC.

If I understand right, your implementation would underestimate the Fisher by (stride ^2), since a.size(1) = g.size(1) * stride.
